File: signature.c
#include <stdio.h>

int signature() {
    printf("Function with no arguments\n");
    return 10;
}

float signature(float a) {
    printf("Function with one argument\n");
    return 10 + a;
}

double signature(double a , double b) {
    printf("Function with two arguments\n");
    return 10 + a + b;
}

int main() {
    // Write C code here
int result1 = signature();
float result2 = signature(1.5);
double result3 = signature(1.5,2.5);

printf("result 1 %d\n ", result1);
printf("result 2 %f\n ", result2);
printf("result 3 %lf\n ", result3);


    return 0;
}

File: signature.py
# Function signature
#   Number of parameters / arguments
#   Order of arguments
#   Data type of arguments
#   Return type

class overloading:
    def signature(self) -> int:
        print("Function with no arguments")
        return 10

    def signature(self, a) -> int:
        print("Function with one argument")
        return 10 + a

    def signature(self, a , b) -> int:
        print("Function with two arguments")
        return 10 + a + b

result1 = overloading()
print(result1.signature())
print(result1.signature(5))
print(result1.signature(10, 20))